src/chapters/guacamole-rest-api.xml

```diff
@@ -0,0 +1,836 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<chapter xml:id="guacamole-rest-api" xmlns="http://docbook.org/ns/docbook"
+    version="5.0" xml:lang="en" xmlns:xl="http://www.w3.org/1999/xlink"
+    xmlns:xi="http://www.w3.org/2001/XInclude">
+    <title>Guacamole REST API</title>
+    <indexterm>
+        <primary>API</primary>
+        <secondary>JavaScript</secondary>
+    </indexterm>
+    <indexterm>
+        <primary>guacamole-rest-api</primary>
+    </indexterm>
+    <para>
+        There are many ways to communicate with the Guacamole protocol, but it is
+        also beneficial to understand how the API works. This allows you to have the
+        flexibility of incorporate Guacamole within your application
+        without having to worry about which language to use or what packages
+        to download.
+    </para>
+
+
+    <section xml:id="base_url">
+        <title>Base URL</title>
+        <para>There are four things you will need to set the base url before
+        you can start talking to the endpoints. </para>
+
+        <itemizedlist>
+            <listitem>
+                <para>Method</para>
+                <emphasis>
+                Protocol your Guacamole instance is using e.g. http, https
+                </emphasis>
+            </listitem>
+            <listitem>
+                <para>Hostname</para>
+                <emphasis>
+                Where your Guacamole is hosted e.g. guacamole.example.com
+                </emphasis>
+            </listitem>
+            <listitem>
+                <para>URL Path</para>
+                <emphasis>
+                The pathway to your Guacamole (where you can see the login page) e.g /guacamole
+                </emphasis>
+            </listitem>
+            <listitem>
+                <para>Data Source</para>
+                <emphasis>
+                Database your Guacamole is running on e.g. mysql, postgresql
```

Comment:
This will not necessarily be a database. Any extension can choose to provide data, and the identifier of the authentication provider that serves as the root object for reaching that data will be the value used for data source.